examples/CURL/CMakeLists.txt

```diff
@@ -9,6 +9,9 @@ include("../common.cmake")
 
 project(download-curl)
 
+hunter_add_package(ZLIB)
+find_package(ZLIB CONFIG REQUIRED)
```

Comment:
The CURL package should export the ZLIB dependency right? https://github.com/cpp-pm/curl/blob/hunter-7.74.0/CMake/Config.cmake.in#L8

Reply:
I'll do whatever you recommend, however:

1. Previous curl produced libraries which didn't seem to need linking with zlib, despite that I don't think I changed any defaults. Maybe upstream curl changed?
2. I would be surprised if hunter packages export their dependencies. What if someone wants to combine a local zlib with hunter curl, for example? Equally, if that's the convention, best stick to it in my book. If I do need to export the zlib dependency, what is the recommended best technique?

Reply:
Typically Hunter packages try to avoid using any system libraries in their dependency tree. The best way to fix this would be to modify the cURL package to export its dependency on ZLIB, but that should already be happening:

https://github.com/cpp-pm/curl/blob/hunter-7.74.0/CMakeLists.txt#L688 - the package is added here
https://github.com/cpp-pm/curl/blob/hunter-7.74.0/CMakeLists.txt#L696 - the package is linked correctly here
https://github.com/cpp-pm/curl/blob/hunter-7.74.0/CMake/Config.cmake.in#L8 - the ZLIB dependency is *incorrectly* exported here, since the variable is `USE_ZLIB` not `HAVE_ZLIB` (https://github.com/cpp-pm/curl/blob/hunter-7.74.0/CMakeLists.txt#L685)

So just changing HAVE to USE in that config file above should fix this issue and allow to remove this add package call here.

Reply:
Not quite fixed as you suggest, but essentially the same thing.